### ISSUE TITLE
500's in erchef on upgrading chef-server

### ISSUE THREAD

**Issue by PrajaktaPurohit** [chef/chef-server#1947]
### Chef Server Version
Seen from 13.1.25 (13.1.13) not seen in 13.0.54(13.0.11)

### Platform Details
ubuntu-18.04

### Configuration
Tiered with external Elasticsearch

### Expected Result:
No 500's with sel_conn_closed

### Actual Result:

```
{<<"method=PUT; path=/organizations/foo-org/data/foo-data-bag-name/foo-data-bag-item;
status=500; ">>,{error,{error,{badmatch,{error,{error,sel_conn_closed}}},
[{oc_chef_wm_base,update_from_json,4,[{file,"/var/cache/omnibus/chef-
server/src/oc_erchef/_build/default/lib/oc_chef_wm/src/oc_chef_wm_base.erl"},{line,1111}]},
{webmachine_resource,resource_call,3,[{file,"/var/cache/omnibus/chef-
server/src/oc_erchef/_build/default/lib/webmachine/src/webmachine_resource.erl"},{line,203}]},
{webmachine_resource,do,3,[{file,"/var/cache/omnibus/chef-
server/src/oc_erchef/_build/default/lib/webmachine/src/webmachine_resource.erl"},{line,148}]},
{webmachine_decision_core,resource_call,1,[{file,"/var/cache/omnibus/chef-
server/src/oc_erchef/_build/default/lib/webmachine/src/webmachine_decision_core.erl"},{line,47}]},
{webmachine_decision_core,accept_helper,1,[{file,"/var/cache/omnibus/chef-
server/src/oc_erchef/_build/default/lib/webmachine/src/webmachine_decision_core.erl"},
{line,613}]},{webmachine_decision_core,decision,1,[{file,"/var/cache/omnibus/chef-
server/src/oc_erchef/_build/default/lib/webmachine/src/webmachine_decision_core.erl"},
{line,518}]},{webmachine_decision_core,handle_request,2,[{file,"/var/cache/omnibus/chef-
server/src/oc_erchef/_build/default/lib/webmachine/src/webmachine_decision_core.erl"},{line,32}]},
{webmachine_mochiweb,loop,2,[{file,"/var/cache/omnibus/chef-
server/src/oc_erchef/_build/default/lib/webmachine/src/webmachine_mochiweb.erl"},{line,106}]}]}}}
```

This request does get retried and eventually succeed.

The error is also seen on other endpoints - /nodes, /search

**Comment by markan** [comment]
As a note, after a little bit of debugging, we saw that error is from the ibrowse library, maybe here, (https://github.com/cmullaparthi/ibrowse/blob/555f7074c35e5afe16efb2d58ef0c3d9aa1d8657/src/ibrowse.erl#L385). Ibrowse was upgraded between 13.0 and 13.1; it's not clear whether something is broken in ibrowse or our usage isn't quite right anymore, but with this being seen with both elasticsearch and solr (#1973) connections it's most likely something on the erchef side.

**Comment by pdohertybcov** [comment]
I was hit by this issue last week.  I upgraded to Chef Infra Server 13.2.0 on April 27th, and ran that configuration without issues until Friday July 17th, when something unknown happened and suddenly I started seeing thousands of errors like the one in this ticket, and things nearly collapsed.  I saw chef-client runs failing about 20% of the time, and the ones that succeeded, all had 500 server errors in them, but would retry and eventually succeed.

After trying a dozen other things, with great guidance from Chef support, I  downgraded the servers to 13.0.16, and everything got instantly better.

Given that other users aren't seeing major issues with 13.2.0, it's likely there is something specific in my environment that triggered the issue, but I have no idea what it is, and it didn't manifest for nearly 3 months of using that version.

My best guess as to the triggering event was the nightly DB backup happened (Amazon RDS). The timing of the backup and the start of the rapid increase in errors line up.  Although this nightly backup never caused problems in the previous 3 months of using 13.2.0, or in the years before that.
